### PR TITLE
[region-isolation] Incremental diagnostic updates

### PR DESCRIPTION
**Pull request by gottesmm**
Just doing a little development on main of the diagnostics as I change them. This just makes it easier to keep other patches in sync with this work.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

File: include/swift/AST/DiagnosticsSIL.def [hunk 1]
```diff
@@ -944,39 +944,25 @@ ERROR(regionbasedisolation_unknown_pattern, none,
...
       (Identifier))
 NOTE(regionbasedisolation_type_is_non_sendable, none,
       "%0 is a non-Sendable type",
       (Type))
+ERROR(regionbasedisolation_type_transfer_yields_race, none,
+      "sending value of non-Sendable type %0 risks causing data races",
+      (Type))
+
+NOTE(regionbasedisolation_type_use_after_transfer, none,
+      "sending value of non-Sendable type %0 to %1 callee risks causing data races between %1 and local %2 uses",
```
Comment (hborla) [inline include/swift/AST/DiagnosticsSIL.def:958]: Can you please avoid introducing new uses of the word `callee` in user-facing diagnostics? It doesn't look like this language was used in the old set of diagnostics.

Comment (gottesmm) [reply include/swift/AST/DiagnosticsSIL.def:958]: I can change it to use the word function.

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
linux failure is unrelated.

**Comment by gottesmm** [comment]
@swift-ci smoke test linux platform

**Pull request merged**